### PR TITLE
Import serverless

### PR DESCRIPTION
**Pull request by AlCutter**
This PR promotes the log specific parts of the `/serverless` directory from `google/trillian-examples` into its own repo.

Notable differences:
 - The `clone2serverless` tool has not been imported; it depends on the `clone` package in `trillian-examples`.
 - The `deploy/github/distributor/...` hierarchy has not been imported.
 - `md-lint` issues have been fixed in the `README` file.

### CODE REVIEW

**Comment by AlCutter** [comment]
This is a large looking PR, but most of the files don't really need a full/deep re-review since they've already been reviewed in `trillian-examples` PRs.

Reviewing commit-by-commit should give a good idea of what's changed.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
+# Serverless Log
+
+This is some experimental tooling which allows for the maintenance and querying
+of a log represented by tiles & files. These tools are built upon components
+from the Trillian repo (e.g. compact ranges), but *do not* use the Trillian Log
+service itself.
```
Comment (jiggoha) [inline README.md:6]: Is the second sentence here still true? Didn't compact ranges get moved to a separate repo? If so, maybe it's more clear to not mention Trillian here at all.

Comment (jiggoha) [reply README.md:6]: Also ok with doing this in a separate PR.

**Review by jiggoha — CHANGES_REQUESTED** [review]
What's the plan for making further progress on the repo split doc? I see that this is moving everything related to serverless to this repo. Then I assume we will further break things out from here to the other repos detailed in the doc?

**Comment by AlCutter** [comment]
> What's the plan for making further progress on the repo split doc? I see that this is moving everything related to serverless to this repo. Then I assume we will further break things out from here to the other repos detailed in the doc?

Exactly, yup.

**Comment by jiggoha** [comment]
> > What's the plan for making further progress on the repo split doc? I see that this is moving everything related to serverless to this repo. Then I assume we will further break things out from here to the other repos detailed in the doc?
> 
> Exactly, yup.

Awesome! Really happy this is happening. As you said, a few logs depend on this now, so it's great to tidy it up to its own thing.

**Pull request merged**